lib/src/signed_video_h26x_auth.c

```diff
@@ -1065,6 +1065,25 @@ signed_video_add_h26x_nalu(signed_video_t *self, const uint8_t *nalu_data, size_
         DEBUG_LOG("No cryptographic information found in SEI. Using default hash algo");
         self->validation_flags.hash_algo_known = true;
       }
+      // If the NALU is golden SEI, proceed with the validation process.
+      if (nalu.is_golden_sei) {
+        prepare_for_validation(self);
+        // Check the status of the verified signature hash for the GOP info.
+        prepare_for_validation(self);
+        switch (self->gop_info->verified_signature_hash) {
+          case 1:
+            nalu_list->first_item->validation_status = '.';
+            break;
+          case 0:
+            nalu_list->first_item->validation_status = 'N';
```

Comment:
There is a flaw in the code if we end up here. If the golden SEI is not authentic the library will report SV_AUTH_RESULT_SIGNATURE_PRESENT and move on.
So, if I make a fake recording and sign it with my own private key, and then replace the video content of another recording using a golden SEI, and also change the public key in that golden SEI, the verification of the golden SEI will fail, but the video will be validated as authentic since it using the tampered public key.